addons/l10n_cl/models/account_move.py

```diff
@@ -273,9 +272,20 @@ def _l10n_cl_get_withholdings(self):
                 'tax_percent': abs(line.tax_line_id.amount),
                 'tax_amount_currency': self.currency_id.round(abs(line.amount_currency)),
                 'tax_amount': self.currency_id.round(abs(line.balance))} for line in self.line_ids.filtered(
-            lambda x: x.tax_group_id.id in [self.env.ref(f'account.{cid}_tax_group_ila').id,
-                                            self.env.ref(f'account.{cid}_tax_group_retenciones').id])]
+            lambda x: x.tax_group_id.id in (
+                self._l10n_cl_get_tax_groups_by_type('ila') + self._l10n_cl_get_tax_groups_by_type('retenciones')).ids)]
         return tax
 
     def _float_repr_float_round(self, value, decimal_places):
         return float_repr(float_round(value, decimal_places), decimal_places)
+
+    def _l10n_cl_get_tax_groups_by_type(self, tax_type):
+        tax_groups = self.env['account.tax.group']
+        companies = self.company_id.parent_ids
+        for company in companies:
+            cid = company.id
+            tax_group = self.env.ref(f'account.{cid}_tax_group_{tax_type}', raise_if_not_found=False)
```

Comment:
self.env['chart.template'].ref('') already has the branch logic?

Reply:
@jco-odoo Indeed. I have updated the fix.

Thanks.